syncer/syncer.go

```diff
@@ -0,0 +1,141 @@
+package syncer
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/0xPolygon/polygon-edge/blockchain"
+	"github.com/0xPolygon/polygon-edge/helper/progress"
+	"github.com/0xPolygon/polygon-edge/syncer/peers"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/hashicorp/go-hclog"
+)
+
+const (
+	LoggerName = "syncer"
+)
+
+type Syncer interface {
+	Start()
+	GetSyncProgression() *progress.Progression
+	BestPeer() *peers.BestPeer
```

Comment:
Why is this an exported method? Who needs to know about the best peer apart from the actual syncing protocol?